### DISCUSSION TITLE
pCloud and KeePassDX are not friends (yet)

### DISCUSSION THREAD

**Discussion by kud** [Kunzisoft/KeePassDX#1869] [Ideas]
Hey,

Hope you're doing well.

First of all, I would like to thank the developers of this project. I really appreciate the app. I used to use KeePass2Android, but this one is much more modern to me, and I'm loving it.

However, there is a little issue for me. Let me explain.

How I use KeepassDX and pCloud for now:
-  I've got my database on pCloud.
-  I then open pCloud on Android.
-  I open KeePass DX.
-  I search for the file via the file system with the pCloud drive.
-  I open the right file.
-  As long as pCloud is active, I can still use it.
-  Once pCloud decides, for security reasons, to close the connection, I can still use my file in KeePass DX in read-only mode.
-  If I want to write it down, I need to first re-open pCloud and also re-open my file in KeePass DX, as I feel the symlink is not the same anymore.

I used to use KeePass2Android, which had better support because it created a third-party connection with pCloud, so it did not need the pCloud Android app itself, just the API.

I don't know if it's possible to trigger pCloud to keep the app active in the background while I use the file there?

Another possibility, do you think it could be possible to do like KeePass2Android and have an integration for pCloud to use it via the API?

Thank you so much for reading this,

Note: I'm also discussing with pCloud team to try to work on the way they keep the app active when a file is currently in use.

**Comment by stravz** [comment]
I experience the same behavior.

**Reply by H234598** [reply]
Same here. Extremly unpleasant as pCloud is very important.

**Comment by J-Jamet** [comment]
The topic of cloud applications has already been mentionned many times.

The goal of KeePassDX is precisely not to be dependent on third-party APIs that promote non-free protocols or services. If pCloud has decided this behavior with the Storage Access Framework, then the pCloud application must be modified.

Cloud APIs in Android are only a way for companies to force direct integrations of their services while a standardized solution exists. What I don't want in a libre application.

To overcome these integration problems and due to the request of the community, I am creating a separate file management application that will contain third-party APIs by plugin.

Please read in detail the wiki that has been written on the subject.

Wiki : https://github.com/Kunzisoft/KeePassDX/wiki/File-Manager-and-Sync
Linked issues : https://github.com/Kunzisoft/KeePassDX/issues/342
https://github.com/Kunzisoft/FileSync/issues/6
